### PR TITLE
Applies new fixes for the homepage and unit page testimonial sliders

### PR DESCRIPTION
**Pull request by jkachel**
### What are the relevant tickets?

Closes mitodl/hq#4644
Closes mitodl/hq#4622

### Description (What does it do?)

Implements a bunch of styling changes to bring the testimonials inline with the Figma designs. There's a lot of them but the gist of it is:
- We now break at `md` and `sm` so there's different styling for tablet users and mobile/phone users
- Testimonial text is truncated and clamped to a specific number of lines/characters per breakpoint
- On the homepage, we eliminate the marketing card image at `md` or lower
- On the homepage, we eliminate the attestant avatar at `sm` or lower
- The avatar image is now _optional_ - we use a Remixicon icon if there's no avatar
- The homepage attestation text has a unique font size - it doesn't match anything we're using currently

Here's some tables to summarize some of the changes as well. On the unit page:
| | `sm` | `md` | `lg` |
|---|---|---|---|
| Attestation max chars | 350 | 350 | 350 |
| Attestation clamp lines | 8 | 6 | 4 |
| Attestation font | `subtitle2` | `subtitle1` | `h5` |
| Attestant title clamp lines | 2 | 2 | 2 |

On the homepage:
| | `sm` | `md` | `lg` |
|---|---|---|---|
| Attestation max chars | 350 | 350 | 350 |
| Attestation clamp lines | 11 | 7 | 7 |
| Attestation font | `subtitle1` | 20px | 20px |
| Attestant title clamp lines | 2 | 2 | 2 |
| Attestant avatar | _No_ | Yes | Yes |

### Screenshots (if appropriate):

*Homepage*

Desktop: long attestation, long attestant title
Screenshot 2024-06-20 at 2 34 46 PM

Desktop: no avatar
Screenshot 2024-06-20 at 2 34 27 PM

Tablet: long attestation, long attestant title
Screenshot 2024-06-20 at 2 35 29 PM

Tablet: long attestation, no avatar
Screenshot 2024-06-20 at 2 37 25 PM

Mobile:
Screenshot 2024-06-20 at 2 39 30 PM

*Unit Page*

Desktop: long attestation, long attestant title
Screenshot 2024-06-20 at 2 42 54 PM

Desktop: no avatar
Screenshot 2024-06-20 at 2 42 48 PM

Tablet: long attestation, long attestant title
Screenshot 2024-06-20 at 2 42 00 PM

Tablet: no avatar:
Screenshot 2024-06-20 at 2 42 18 PM

Mobile: long attestation, long attestant title
Screenshot 2024-06-20 at 2 41 25 PM

Mobile: no avatar
Screenshot 2024-06-20 at 2 41 05 PM

### How can this be tested?

Automated tests should pass. 

You should be able to configure an attestation with no avatar image set. In addition, you should be able to clear avatars from attestations. 

The design should look like the design in Figma. (Links are available in the original issues.)

### CODE REVIEW

**Comment by steven-hatch** [comment]
Great work! 2 things: 

1. Can we go 100% width on the name/title section? We can still keep the wrapping text for extreme cases, but 99% should conform into this space. 
image

2. Vertical align text MIDDLE - We are aligning top, but can we vertically align middle (or align-items: center)? Both the Units page and the home page version should do this for the text area. **The Reason:** We are supporting quotes that vary from 100-350 characters. Keeping the text aligned middle will balance the cards, even if not much content is in there. 
Screenshot 2024-06-20 at 5 04 42 PM

<hr>

I have updated the figma with bounding boxes to illustrate both issues. https://www.figma.com/proto/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?page-id=914%3A13944&node-id=7566-164987&viewport=-32419%2C-2178%2C0.43&t=SNYW2zQX7ct02AOq-1&scaling=min-zoom&starting-point-node-id=5348%3A173543&show-proto-sidebar=1

**Comment by pdpinch** [comment]
If it's useful, the testimonials in production now are all ready for production (maybe easier to copy and paste from [this doc](https://docs.google.com/spreadsheets/d/1VEccOViAGQuVpS_XlN7DJa_ZnVv7lzFiEfqVcy08PiE/edit#gid=0)). We're just missing the OCW testimonials.

**Comment by jkachel** [comment]
@steven-hatch fixed. Here's a handful of new screenshots:

Desktop (one with long attestation and attestant title, one with neither so you can see the centering clearly):
Screenshot 2024-06-21 at 7 51 21 AM
Screenshot 2024-06-21 at 7 51 15 AM

Tablet:
Screenshot 2024-06-21 at 7 53 01 AM

Mobile:
Screenshot 2024-06-21 at 7 53 20 AM

**Comment by steven-hatch** [comment]
Looking good! Thank you so much for you effort and flexibility on this! This should be good for Monday, and eager to see it on RC.

File: frontends/mit-open/src/page-components/TestimonialDisplay/TestimonialDisplay.tsx [hunk 1]
```diff
@@ -24,6 +42,7 @@ const QuoteContainer = styled.section(({ theme }) => ({
   marginBottom: "80px",
   [theme.breakpoints.down("md")]: {
     marginBottom: "40px",
+    height: "200px",
```
Comment (jonkafton) [inline frontends/mit-open/src/page-components/TestimonialDisplay/TestimonialDisplay.tsx:45]: Do we want to fix the height for mobile? I'd avoid scroll inside scroll where possible.

image

**Comment by jonkafton** [comment]
An issue with the carousel, not really this PR - At screens smaller than around 680px and above the 600px breakpoint, the testimonial card loses its margin and doesn't look quite right:

image

**Comment by jkachel** [comment]
> An issue with the carousel, not really this PR - At screens smaller than around 680px and above the 600px breakpoint, the testimonial card loses its margin and doesn't look quite right:
> image

I think this is maybe a breakpoint thing - I'm using `md` basically for tablet but maybe we should have a new one. @steven-hatch I think you mentioned something along these lines. Playing with it in the responsive view in Firefox, it seems like we should drop into the smallest view maybe around 700px, and tablet should be between 1280 > tablet > 700?

File: frontends/mit-open/src/page-components/TestimonialDisplay/TestimonialDisplay.tsx (hunk 1 above)
Comment (jkachel) [reply frontends/mit-open/src/page-components/TestimonialDisplay/TestimonialDisplay.tsx:45]: Should be fixed in https://github.com/mitodl/mit-open/pull/1131/commits/062a703348e430bcce1c389a15a821e064f74fe0 - think I nerfed this accidentally during a recent rebase.

**Review by jonkafton — APPROVED** [review]

**Pull request merged**